src/_version_check.sh

```diff
@@ -0,0 +1,187 @@
+        stable_date=$(date -d $(wget --header="Authorization: JWT "${JWT_TOKEN} -O - "https://hub.docker.com/v2/repositories/plextrac/plextracapi/tags/stable" -q | jq -r .tag_last_pushed) +%s)
+        # Get date for the latest version tag was pushed
+        latest_date=$(date -d $(wget --header="Authorization: JWT "${JWT_TOKEN} -O - "https://hub.docker.com/v2/repositories/plextrac/plextracapi/tags/$latest_ver" -q | jq -r .tag_last_pushed) +%s)
+        
+        ## LOGIC: LATEST_STABLE
+        # IF LATEST_STABLE <= 2.0
+        if (( $(echo "$latest_ver <= $breaking_ver" | bc -l) ))
+          then 
+            debug "$breaking_ver not publically available. Updating normally without warning"
+            contiguous_update=false
+          
+          # IF LATEST_STABLE >= 2.1
+          else
+            debug "Stable version is greater than $breaking_ver. Running contiguous update"
+            contiguous_update=true
+        fi
+      # If the JWT token is empty for on-prem envs
+      else
+        contiguous_update=false
+        error "Unable to validate versioned images from DockerHub. Likely On-prem or Air-gapped"
+        msg "-------"
+        error "Beginning with version 2.0, PlexTrac is going to require contiguous updates to ensure code migrations are successful and enable us to continue to move forward with improving the platform. We recommend updating to the next minor version available compared to the running version $running_backend_version"
+        error "Are you sure you want to update to $UPGRADE_STRATEGY? (y/n)"
+        get_user_approval hardcheck
+    fi
+
+    upstream_tags=()
+    page=1
+    if [ "$contiguous_update" = true ]
+      then
+        # Get upstream tag list
+        debug "Looking for Running version $running_ver or Breaking version $breaking_ver"
+        while [ $page -lt 600 ]
+          do
+            # Get the available versions from DockerHub and save to array
+            if [[ $(echo "${upstream_tags[@]}" | grep "$breaking_ver" || true) ]]
+              then
+                debug "Found breaking version $breaking_ver"; break;
+            elif [[ $(echo "${upstream_tags[@]}" | grep "$running_ver" || true) ]]
+              then
+                debug "Found running version $running_backend_version"; break;
+            else
+              upstream_tags+=(`wget --header="Authorization: JWT "${JWT_TOKEN} -O - "https://hub.docker.com/v2/repositories/plextrac/plextracapi/tags/?page=$page&page_size=1000" -q | jq -r .results[].name | grep -E '(^[0-9]\.[0-9]*$)' || true`)
+            fi
+            page=$[$page+1]
+        done
+        # Compare stable vs latest_tag and if stable is older than the newest tag, then remove the newest tag to ensure we don't update past stable
+        if [ $stable_date -le $latest_date ]; then
+          debug "Stable is older than latest version"
+          # If stable is older, remove latest from upgrade path
+          for i in ${!upstream_tags[@]}
```

Comment:
if upstream_tags is a sorted list, you could just pop the first item rather than iterate through the entire list looking for it. Based on what I'm reading, I _think_ it should be sorted

Reply:
This is removed pending other changes to the upstream DockerHUB repo for tighter validation